### DISCUSSION TITLE
Disambiguation for tokenization

### DISCUSSION THREAD

**Discussion by ozwiz** [explosion/spaCy#425] [New Features & Project Ideas]
First of all thanks for your great work.
I have read many issues posted here about tokenization. It is a tough task. For example "Adam's" may mean "of Adam" or "Adam is".
Would you consider any disambiguation step that will merge or split "'s" automatically based on later functions?

**Comment by ines** [comment]
Quick update: This might be a nice use case for the new [custom processing pipeline components](https://alpha.spacy.io/usage/processing-pipelines#custom-components) and [extension attributes](https://spacy.io/usage/processing-pipelines#custom-components-attributes) introduced in v2.0!

**Comment by naveenjafer** [comment]
This seems to be a really old thread, but I would like to take it up if there is still a need for this. @ines is this relevant as of today? If there is a plan to proceed with this, I might really benefit from a few more functional examples of the same.

**Comment by naveenjafer** [comment]
Hi @ines @honnibal is the idea to keep multiple possible tokenizations for the same sentence where applicable and evaluate the one that is grammatically most probable in later stages of the pipeline before committing to one of them?
